### ISSUE TITLE
SingularField should consider variable modifier

### ISSUE THREAD

**Issue by Belle-PL** [pmd/pmd#3853]
**Affects PMD Version:6.42.0** 

**Rule:SingularField**

Please provide the rule name and a link to the rule documentation:
<https://pmd.github.io/latest/pmd_rules_java_design.html#singularfield>

**Description:**
Hi, I found a false negative about the rule SingularField. In the following code example, variable `x` actually should be defined only in method `bar` because it is only used in this method. Hence, PMD should report a warning at line 1 and I think this is a false negative.

**Code Sample demonstrating the issue:**
```
private static int x;  // should report a warning here
int bar(int y) {
    x = y + 5;
    return x;
}
```

**Expected outcome: A warning**

PMD should report a violation at line 1, but doesn't. This is a false-negative.

**Running PMD through:** *[Maven]*

**Comment by oowekyala** [comment]
I don't think this is true @adangel, the field is never read before it's assigned. The semantics are the same as
```
int bar(int y) {
   int x = y + 5;
   return x;
}
```
and this code doesn't have a race condition

**Comment by adangel** [comment]
Ah, yes, my bad. I thought x would be used on the right side as well (why else would you make it static :))
